cms/src/components/images/ImageEditor.vue

```diff
@@ -0,0 +1,186 @@
+const nameInput = ref<typeof LInput | undefined>(undefined);
+const descriptionInput = ref<typeof LInput | undefined>(undefined);
+const uploadInput = ref<typeof HTMLInputElement | undefined>(undefined);
+
+const presets = [
+    { label: "default", value: "default" },
+    { label: "photo", value: "photo" },
+    { label: "picture", value: "picture" },
+    { label: "drawing", value: "drawing" },
+    { label: "icon", value: "icon" },
+    { label: "text", value: "text" },
+];
+const selectedPreset = ref(presets[1].value);
+
+// Methods
+const save = () => {
+    const raw = toRaw(image.value);
+    delete raw.uploadData; // do not re-save upload data as this will trigger a re-render of the image in the API.
+    db.upsert<ImageDto>(toRaw(image.value));
+};
+
+const showFilePicker = () => {
+    // @ts-ignore
+    uploadInput.value!.showPicker();
+};
+
+// Read files into Buffer
+const upload = () => {
+    // @ts-ignore
+    if (!uploadInput.value!.files || uploadInput.value!.files.length == 0) return;
+
+    const reader = new FileReader();
+    // @ts-ignore
+    const file = uploadInput.value!.files[0];
+
+    reader.onload = (e) => {
+        if (!image.value.uploadData) image.value.uploadData = [] as ImageUploadDto[];
+        const fileData = e.target!.result as ArrayBuffer;
+
+        if (fileData.byteLength > globalConfigStore.maxUploadFileSize) {
+            addNotification({
+                title: `Invalid image file size`,
+                description: `Image file size is larger than the maximum allowed size of ${maxUploadFileSize.value}MB`,
+                state: "error",
+            });
+            return;
+        }
+
+        // Save the file data
+        const raw = toRaw(image.value);
+        raw.uploadData = [
+            {
+                fileData,
+                preset: selectedPreset.value,
+                filename: file.name,
+            },
+        ]; // do not re-save previous upload data as this will trigger a re-render of the image in the API.
+        db.upsert<ImageDto>(toRaw(image.value));
+    };
+
+    reader.readAsArrayBuffer(file);
+
+    // Reset the file input
+    // @ts-ignore
+    uploadInput.value!.value = "";
+};
+
+const removeFile = (filename: string) => {
+    image.value.files = image.value.files
+        .filter((f) => f.filename !== filename)
+        .map((f) => toRaw(f));
+    save();
+};
+</script>
+
+<template>
+    <div class="flex-col">
+        <label class="text-xs italic text-zinc-500">Changes are applied immediately</label>
+        <div>
+            <LInput
+                ref="nameInput"
+                v-model="image.name"
+                name="fileName"
+                label="Image name"
+                class="mb-2 flex-1"
+                @change="save"
+                data-test="image-name"
+            />
+            <div class="mb-2 flex items-end gap-4">
+                <LSelect
+                    name="preset"
+                    label="Upload preset"
```

Comment:
I find the "upload preset" field not super clear - I guess it does some specific optimizations related to the type? But not sure if editors will understand it, for example the difference between "photo", "picture" and "icon" (I also don't understand that 😅).

Reply:
The preset is something that comes from the Sharp image processing library. Perhaps we should rename the "Upload preset" to something like "What type of image are you uploading?" and rename the options to:
- Photo
- Drawing
- Drawing with text
- Other

Photo will be the most common, and therefore is the default. (Options like "icon" can be removed as it is not really applicable here)

Reply:
I think that will help 👍 Still a bit torn whether it should be as prominent as it is. It can be tricky but maybe hide it behind an "advanced settings" or something. We can safely merge it but we should think a bit about this before onboarding people that we don't overload the UI with things that most of the time are not relevant (thus returning to the current AC admin situation).

Reply:
Happy to remove the dropdown and leave it on default "photo". Effect will be that some images will be stored in less-than-ideal compression, but that will be very few as I see it at the moment.

Reply:
> Happy to remove the dropdown and leave it on default "photo". Effect will be that some images will be stored in less-than-ideal compression, but that will be very few as I see it at the moment.

I think that would be nice 👍 If we need it we can bring it back in the future but think a bit more about the ergonomics for editors/less technical people, like having them explicitly select before uploading "is this a normal image or image with text".